engine/src/flutter/testing/ios/IosUnitTests/App/AppDelegate.m

```diff
@@ -30,4 +30,8 @@ - (void)applicationDidBecomeActive:(UIApplication*)application {
 - (void)applicationWillTerminate:(UIApplication*)application {
 }
 
+- (FlutterEngine*)grabLaunchEngine {
+  return _launchEngine;
```

Comment:
is this added by mistake? (it's different implementation from the `grabLaunchEngine` inside `FlutterAppDelegate.mm`)

Reply:
Nope this isn't a mistake.  This is for the ios unit tests.  FlutterViewController is reading from the application delegate which can't be swapped at runtime.  This is how we can mock out that behavior.

Reply:
^^ unresolving this in case you missed it.

Reply:
Ah `_launchEngine` is of type `FlutterEngine`, not `FlutterLaunchEngine`. This is *VERY* confusing naming!

For sanity of future readers, I'd like us to revisit the renaming proposal

Reply:
I don't think the `FlutterLaunchEngine` concept is confusing, otherwise that would have shown up in the design doc.  I think you are getting tripped up by the weird stuff we had to do to test the FlutterAppDelegate which was a bit confusing.  I'm renamed that property to `mockLaunchEngine` and added a comment the explained why it behaves differently for the tests.